src/GUI/GamePanel.java

```diff
@@ -0,0 +1,79 @@
+package GUI;
+
+import Gameplay.*;
+
+import javax.swing.*;
+import java.awt.*;
+import java.io.IOException;
+
+public class GamePanel extends JPanel {
+
+    protected static int XOffset = 100;
+    public static Point deckPosition = new Point(620, 20);
+    public static Point tableauPosition = new Point(20, 150);
+    public static int tableauOffset = 100;
+    private static Deck deck;
+    private static WastePile wastePile;
+    private static Foundations[] foundations;
+    private static Tableau[] tableaus;
```

Comment:
```suggestion
    private Deck deck;
    private WastePile wastePile;
    private Foundations[] foundations;
    private Tableau[] tableaus;
```

I see no reason for these to be `static`.

Static = single instance (global)
Non-static = bound to current instance (`this`)

I imagine the `deck`, `wastePile`, `foundations` and `tableaus` belong to an instance of `GamePanel`.
If they do not, this does not respect OOP encapsulation.

Furthermore, the getters for these make no sense if static.
You acquire these from the `GamePanel`. instance

Assuming these and the getters were no longer static, to get the `deck` of a `GamePanel`.
```java
void somewhereInCode(..., GamePanel panel) {
    var deck = panel.getDeck();
}
```

Just pass the instance along to the interacting classes.